twine/exceptions.py

```diff
@@ -73,3 +73,15 @@ class InvalidSigningConfiguration(TwineException):
     """Both the sign and identity parameters must be present."""
 
     pass
+
+
+class InvalidConfiguration(TwineException, TypeError):
```

Comment:
I'm not convinced these need to doubly inherit. We still don't have a right and proper stable public API.

Reply:
It doesn't hurt anything except maintainability. These have to go on forever with this inheritance pattern. It's also not clear if the following is ideal behaviour:

```py
try:
    some_other_function(too, few, args)
    twine.api.something_that_raises_invalidconfiguration(...)
except InvalidConfiguration:
    print('Twine broken')
```

yes the user is writing code that's not great, but we're also then obfuscating exactly where that came from because `InvalidConfiguration` would catch the TypeError from the first call. And while we can all agree this code isn't ideal, this is a common pattern in the community, so I think it's an anti-pattern for us to go out of our way to break it, even if it means breaking an un-stable semi-public API that it appears very few are even attempting to use today.

Reply:
OK, I've removed the commit that added them.